iroh-net/src/dns.rs

```diff
@@ -67,7 +67,11 @@ fn create_default_resolver() -> Result<TokioAsyncResolver> {
         }
     }
 
-    // lookup IPv4 and IPv6 in parallel
+    // ensure multiple name servers are queried concurrently
+    options.num_concurrent_reqs = 3;
```

Comment:
Sorry, only just realised this comment.  It queries multiple DNS servers concurrently, but we typically only have a single DNS server.  We want (I think) to be robust in case the UDP packet to this server is lost and thus do multiple concurrent requests to the same server, but with staggered starts so that you don't always pay the price of the multiple requests if your DNS server is fast enough.